runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/ChatCommandsPlugin.java

```diff
@@ -1448,16 +1448,7 @@ private HiscoreLookup getCorrectLookupFor(final ChatMessage chatMessage)
 			return new HiscoreLookup(localPlayer.getName(), hiscoreEndpoint);
 		}
 
-		// Public chat on a leagues world is always league hiscores, regardless of icon
-		if (chatMessage.getType() == ChatMessageType.PUBLICCHAT || chatMessage.getType() == ChatMessageType.MODCHAT)
-		{
-			if (client.getWorldType().contains(WorldType.LEAGUE))
-			{
-				return new HiscoreLookup(player, HiscoreEndpoint.LEAGUE);
-			}
-		}
-
-		// Get ironman status from their icon in chat
+		// Get ironman status from their icon in chat, this handles leagues too
```

Comment:
I'm pretty sure there doesn't need to be extra code for leagues worlds since getHiscoreEndpointByName would handle this case anyways but if someone disagrees I can re-add